lib/MusicBrainz/Server/Form/Utils.pm

```diff
@@ -231,6 +232,25 @@ sub indentation {
     return "\N{NO-BREAK SPACE}" x (3 * $level);
 }
 
+sub validate_username {
+    my ($self) = @_;
+
+    my $username = $self->value;
+    my $previous_username = $self->init_value;
+
+    if (defined $username) {
```

Comment:
Hm, don't trust me too much on that, it's possible it runs all the validation handlers even if it fails the `required` one.